expression/builtin_cast_vec.go

```diff
@@ -38,3 +39,40 @@ func (b *builtinCastIntAsIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 func (b *builtinCastIntAsIntSig) vectorized() bool {
 	return true
 }
+
+func (b *builtinCastIntAsRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeFloat64(n, false)
+
+	i64s := buf.Int64s()
+	rs := result.Float64s()
+
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.SetNull(i, true)
```

Comment:
To calculate result.IsNull(), maybe it's better to use result.MergeNulls(). Can you do some benchmark?